zrcs/zrc-2.md

```diff
@@ -0,0 +1,412 @@
+| eventName | `Error`                    | Checking operator is not successful. | TBA.                                                                                                      |
+
+
+#### 12. DefaultOperators
+
+```ocaml
+(* @dev: Returns the list of default operators. These addresses are operators for all token holders. *)
+transition DefaultOperators()
+```
+
+|           | Name                       | Description                             | Event Parameters                                                                                                                                                                                                               |
+| --------- | -------------------------- | --------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| eventName | `DefaultOperatorsSuccess` | Listing default operators is successful.     | `list`: `List ByStr20` |
+
+
+#### 13. Transfer
+
+```ocaml
+(* @dev: Move a given amount of tokens from one address another.       *)
+(* @param to:     Address of the recipient whose balance is increased. *)
+(* @param amount: Number of tokens to be transferred.                  *)
+transition Transfer(to: ByStr20, amount: Uint128)
+```
+
+|        | Name      | Type      | Description                        |
+| ------ | --------- | --------- | ---------------------------------- |
+| @param | `to`      | `ByStr20` | Address of the recipient whose balance is increased. |
+| @param | `amount`  | `Uint128` | Number of tokens to be transferred.                  |
+
+|           | Name                  | Description                 | Event Parameters                                                                                                                                                                                                                                                                        |
+| --------- | --------------------- | --------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| eventName | `TransferSuccess` | Transfering is successful.     | `to`: `ByStr20`, and `amount`: `Uint128`.                                                                            |
+| eventName | `Error`               | Transfering is not successful. | - emit `CodeNotAuthorised` if the transition is called by a user that is not authorised.<br>**NOTE:** Only `tokenOwner` address can invoke this transition. |
```

Comment:
Again, since the `tokenOwner` is implicitly the caller, the error should more appropriately be "out of balance". The sender is _always_ authorized, he just may or may not have sufficient balance.